### PR TITLE
[CVP-1226] operator metadata role update

### PR DESCRIPTION
**Pull request by jsztuka**
This change is primarily made in order to fix following:
https://github.com/operator-framework/community-operators/pull/2014

### CODE REVIEW

**Comment by sonam1412** [comment]
what's the purpose of log.txt ? maybe got added accidently

**Comment by jsztuka** [comment]
> what's the purpose of log.txt ? maybe got added accidently

Yes, it got involved by accident. Thanks for noticing.

File: roles/inject_scorecard_metadata/tasks/main.yml [hunk 1]
```diff
@@ -48,14 +48,17 @@
...
+  register: csv_data
+
+- name: "Set facts for csv_data"
+  set_fact:
+    csv_vars: "{{ csv_data.stdout | from_yaml }}"
 
 - name: "Get spec from csv_vars"
   set_fact:
-    csv_vars_spec: "{{ csv_vars['spec'] | default({}) }}"
+    csv_vars_spec: "{{ (csv_vars | from_yaml)['spec'] | default({}) }}"
```
Comment (sonam1412) [inline roles/inject_scorecard_metadata/tasks/main.yml:61]: Do we have to convert the csv_vars again here when it is already done on line#57 ?

File: roles/parse_operator_bundle/tasks/main.yml [hunk 2]
```diff
@@ -89,39 +89,42 @@
...
   - name: "Determine and set fact for operator specific information - name, pod name, container name and capabilities"
     set_fact:
-      current_csv: "{{ operator_vars.metadata.name }}"
-      operator_pod_name: "{{ operator_vars.spec.install.spec.deployments[0].name }}"
-      operator_container_name: "{{ operator_vars.spec.install.spec.deployments[0].spec.template.spec.containers[0].name }}"
-      operator_capabilities: "{{ operator_vars.metadata.annotations.capabilities }}"
+      current_csv: "{{ (csv_vars | from_yaml).metadata.name }}"
+      operator_pod_name: "{{ (csv_vars | from_yaml).spec.install.spec.deployments[0].name }}"
+      operator_container_name: "{{ (csv_vars | from_yaml).spec.install.spec.deployments[0].spec.template.spec.containers[0].name }}"
+      operator_capabilities: "{{ (csv_vars | from_yaml).metadata.annotations.capabilities }}"
```
Comment (sonam1412) [inline roles/parse_operator_bundle/tasks/main.yml:105]: Do we have to convert the csv_vars again here when it is already done on line#98 ?

File: roles/parse_operator_metadata/tasks/main.yml [hunk 3]
```diff
@@ -118,39 +118,42 @@
...
-    include_vars:
-      file: "{{ csv_path }}"
-      name: operator_vars
+  - name: "Read csv.yaml file"
+    shell: "cat {{ csv_path }}"
+    register: csv_data
+
+  - name: "Set facts for csv_data"
+    set_fact:
+      csv_vars: "{{ csv_data.stdout }}"
```
Comment (sonam1412) [inline roles/parse_operator_metadata/tasks/main.yml:127]: Shouldn't this be converted here and use csv_vars at all other recursive places ?

File: roles/inject_scorecard_metadata/tasks/main.yml (hunk 1 above)
Comment (jsztuka) [reply roles/inject_scorecard_metadata/tasks/main.yml:61]: If we used from_yaml only in the initial part of declaring csv_vars and no further, nested variables of that fact(csv_vars) would be unaccessible by ansible.

File: roles/parse_operator_bundle/tasks/main.yml (hunk 2 above)
Comment (jsztuka) [reply roles/parse_operator_bundle/tasks/main.yml:105]: Same case as mentioned above.

File: roles/parse_operator_metadata/tasks/main.yml (hunk 3 above)
Comment (jsztuka) [reply roles/parse_operator_metadata/tasks/main.yml:127]: Same case as mentioned above.

Comment (sonam1412) [reply roles/parse_operator_metadata/tasks/main.yml:127]: from_yaml is not needed here?

Comment (jsztuka) [reply roles/parse_operator_metadata/tasks/main.yml]: It is, updated.

**Comment by sonam1412** [comment]
LGTM

**Pull request merged**